Structure_Engine/Compute/FEMeshToPanel.cs

```diff
@@ -0,0 +1,112 @@
+ */
+
+
+using BH.oM.Structure.Elements;
+using BH.oM.Geometry;
+using System.Collections.Generic;
+using System.Linq;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Structure
+{
+    public static partial class Compute
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+        [Description("Converts an FEMesh to a Panel, each MeshFace is converted to a Panel with identical properties and fragments.")]
+        [Input("feMesh", "FEMesh to be converted to a Panel.")]
+        [Output("panel", "Panel converted from a FEMesh.")]
+
+        public static List<Panel> FEMeshToPanel(FEMesh feMesh)
+        {
+            if (feMesh == null)
+            {
+                Engine.Reflection.Compute.RecordError("Checks identify null FEMesh");
+                return null;
+            }
+            if (feMesh.Nodes == null)
+            {
+                Engine.Reflection.Compute.RecordError("Checks identify null Nodes");
+                return null;
+            }
+            if (feMesh.Faces == null)
+            {
+                Engine.Reflection.Compute.RecordError("Checks identify null Faces");
+                return null;
+            }
+            foreach (Node node in feMesh.Nodes)
+            {
+                if (node == null)
+                {
+                    Reflection.Compute.RecordError("Checks identify null Node");
+                    return null;
+                }
+            }
+            foreach (FEMeshFace face in feMesh.Faces)
+            {
+                if (face == null)
+                {
+                    Reflection.Compute.RecordError("Checks identify null Face");
+                    return null;
+                }
+            }
+            List<Polyline> polylines = new List<Polyline>();
+            List<Point> points = new List<Point>();
+
+            foreach (FEMeshFace feMeshFace in feMesh.Faces)
+            {
+                foreach (int nodeIndex in feMeshFace.NodeListIndices)
+                {
+                    points.Add(feMesh.Nodes[nodeIndex].Position);
+                    points.Add(feMesh.Nodes.First().Position);
+                }
+                polylines.Add(Geometry.Create.Polyline(points));
+            }
+            List<Panel> panels = new List<Panel>();
+            Panel panel = new Panel();
+            foreach (Polyline polyline in polylines)
+            {
+                if (feMesh.Property != null)
+                {
+                    panel = Create.Panel(polyline, null, feMesh.Property, feMesh.Name);
+                }
+                else
+                {
+                    panel = Create.Panel(polyline, null, null, feMesh.Name);
+                    Reflection.Compute.RecordWarning("FEMeshes don't have any Section Property input");
+                }
```

Comment:
You can create the `Panel` without a `SurfaceProperty` and then check `if (feMesh.Property != null)` whether you assign it or not - then you don't have the `Panel` create method duplicated.